pkg/controller/deployment/deployment_controller.go

```diff
@@ -922,6 +927,21 @@ func copyDeploymentAnnotationsToReplicaSet(deployment *extensions.Deployment, rs
 	return rsAnnotationsChanged
 }
 
+// copyReplicaSetAnnotationsToDeployment copies RS's annotations to deployment's annotations.
+// This action should be done if and only if the deployment is rolling back to this rs.
+// Note that apply and revision annotations are not copied.
```

Comment:
Revision is only updated (to max old revision + 1, if it's larger) in getNewReplicaSet and then copied to deployment (revision in deployment is for reference only and is always copied from new RS to deployment). We don't touch it here.